nats-concepts/jetstream/source_and_mirror.md

```diff
@@ -0,0 +1,69 @@
+# Source and Mirror
+
+When a stream is configured with a `source` or `mirror`, it will automatically and asynchronously replicate messages from the origin stream. There are several options when declaring the configuration.
+
```

Comment:
Though you touch on in it "Expected behavior" I would put right here at the start in this first paragraph something about the replication being "reliable store and forward" kind of replication. To make sure people get that it's reliable so all the messages are replicated (eventually) even in the face of failures (e.g. leaf nodes on partially connected devices).

Reply:
Extended description